apollo-router/src/cache/redis.rs

```diff
@@ -180,22 +189,16 @@ impl RedisCacheStorage {
 
                 let mut result = first.clone();
 
-                match scheme {
-                    "redis" => {
-                        let _ = result.set_scheme("redis-cluster");
-                    }
-                    "rediss" => {
-                        let _ = result.set_scheme("rediss-cluster");
-                    }
-                    other => {
-                        return Err(RedisError::new(
-                            RedisErrorKind::Config,
-                            format!(
-                                "invalid Redis URL scheme, expected 'redis' or 'rediss', got: {}",
-                                other
-                            ),
-                        ))
-                    }
+                if SUPPORTED_REDIS_SCHEMES.contains(&scheme) {
+                    let _ = result.set_scheme(scheme);
```

Comment:
this will be a regression: in the previous code, if we set multiple `redis://` URLs, it resulted in a single `redis-cluster://` URL

Reply:
Good catch. I'll push a change which preserves the behaviour of converting `redis` to `redis-cluster` and `rediss` to `rediss-cluster`

done: 8a6e8bcb838ffeba661076bd8d214e21304e9c29

Reply:
I thought some more about this overnight and concluded that change would still be a regression, since it couldn't handle incompatibilities with Sentinel. So, I've done a more comprehensive change which (I hope) will now address all of the various complexities.
see: 92ec1c187dda3dfb1fe6106fe82d196112e100ca